azblob/zt_url_blob_test.go

```diff
@@ -1970,3 +1970,192 @@ func (s *aztestsSuite) TestDownloadBlockBlobUnexpectedEOF(c *chk.C) {
+	blobURL, _ := getBlockBlobURL(c, containerURL)
+
+	blockBlobUploadResp, err := blobURL.Upload(ctx, bytes.NewReader([]byte("data")), BlobHTTPHeaders{},
+		basicMetadata, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(blockBlobUploadResp.VersionID(), chk.NotNil)
+	versionId1 := blockBlobUploadResp.VersionID()
+
+	blockBlobUploadResp, err = blobURL.Upload(ctx, bytes.NewReader([]byte("updated_data")), BlobHTTPHeaders{},
+		basicMetadata, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(blockBlobUploadResp.VersionID(), chk.NotNil)
+
+	listBlobsResp, err := containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Versions: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobsResp.Segment.BlobItems, chk.HasLen, 2)
+
+	// Deleting previous version snapshot.
+	deleteResp, err := blobURL.WithVersionID(versionId1).Delete(ctx, DeleteSnapshotsOptionNone, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(deleteResp.StatusCode(), chk.Equals, 202)
+
+	listBlobsResp, err = containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Versions: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobsResp.Segment.BlobItems, chk.NotNil)
+	if len(listBlobsResp.Segment.BlobItems) == 0 {
+		c.Fail()
+	}
+}
+
+func (s *aztestsSuite) TestDeleteSpecificBlobVersionWithBlobSAS(c *chk.C) {
+	bsu := getBSU()
+	credential, err := getGenericCredential("")
+	if err != nil {
+		c.Fatal(err)
+	}
+	containerURL, containerName := createNewContainer(c, bsu)
+	defer deleteContainer(c, containerURL)
+	blobURL, blobName := getBlockBlobURL(c, containerURL)
+
+	resp, err := blobURL.Upload(ctx, bytes.NewReader([]byte("data")), BlobHTTPHeaders{},
+		basicMetadata, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	versionId := resp.VersionID()
+	c.Assert(versionId, chk.NotNil)
+
+	resp, err = blobURL.Upload(ctx, bytes.NewReader([]byte("updated_data")), BlobHTTPHeaders{},
+		basicMetadata, BlobAccessConditions{})
+	c.Assert(err, chk.IsNil)
+	c.Assert(resp.VersionID(), chk.NotNil)
+
+	blobParts := NewBlobURLParts(blobURL.URL())
+	blobParts.VersionID = versionId
+	blobParts.SAS, err = BlobSASSignatureValues{
+		Protocol:      SASProtocolHTTPS,
+		ExpiryTime:    time.Now().UTC().Add(1 * time.Hour),
+		ContainerName: containerName,
+		BlobName:      blobName,
+		Permissions:   BlobSASPermissions{Delete: true, DeletePreviousVersion: true}.String(),
+	}.NewSASQueryParameters(credential)
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	sbURL := NewBlockBlobURL(blobParts.URL(), containerURL.client.p)
+	deleteResp, err := sbURL.Delete(ctx, DeleteSnapshotsOptionNone, BlobAccessConditions{})
+	c.Assert(deleteResp, chk.IsNil)
+
+	listBlobResp, err := containerURL.ListBlobsFlatSegment(ctx, Marker{}, ListBlobsSegmentOptions{Details: BlobListingDetails{Versions: true}})
+	c.Assert(err, chk.IsNil)
+	c.Assert(listBlobResp.Version(), chk.Equals, ServiceVersion)
```

Comment:
>c.Assert(listBlobResp.Version(), chk.Equals, ServiceVersion) [](start = 1, length = 60)

a bit confused by this validation, why are we validating the service version?

Reply:
Versioning is supported by service version `2019-12-12` or higher. I used this validation as a way to check/flag that versioning is enabled on the storage account.

Reply:
this will always succeed thought, right? Since the SDK will use the Service Version by default